### PR TITLE
 Add docs for `*args` usage for node definition

### PR DESCRIPTION
**Pull request by noklam**
Signed-off-by: Nok <nok.lam.chan@quantumblack.com>

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: docs/source/nodes_and_pipelines/nodes.md [hunk 1]
```diff
@@ -87,6 +88,9 @@ A syntax describes function inputs and outputs. This syntax allows different Pyt
 
 Any combinations of the above are possible, except nodes of the form `node(f, None, None)` (at least a single input or output must be provided).
 
+## `*args` node functions
+It is common to have functions that take an arbitrary number of inputs. For example, a function that combines multiple dataframes. You can use the *args argument in the node function, while simply declaring the names of the datasets in the node's inputs.
```
Comment (deepyaman) [inline docs/source/nodes_and_pipelines/nodes.md:92]:
```suggestion
It is common to have functions that take an arbitrary number of inputs, like a function that combines multiple dataframes. You can use the `*args` argument in the node function, while simply declaring the names of the datasets in the node's inputs.
```
Just a sentence fragment.

Also, does it make sense to call them by a name, like variadic arguments, or too confusing?

**Review by deepyaman — APPROVED** [review]
Just a minor change request, but will approve to unblock pending that

File: docs/source/nodes_and_pipelines/nodes.md (hunk 1 above)
Comment (noklam) [reply docs/source/nodes_and_pipelines/nodes.md]: I am not sure, variadic isn't a name familiar to me. I did a quick search and seems like Python ecosystem also don't really have a name for it. i.e. `varying number of arguments.` , `non-keyworded variable length argument list`. Varidiac does show up in docsbut in the standard doc they call the section Arbitary argument list" " https://docs.python.org/3/tutorial/controlflow.html#arbitrary-argument-lists"

**Review by merelcht — APPROVED** [review]
LGTM! 👍

**Pull request merged**